src/API.jl

```diff
@@ -977,4 +977,22 @@ Package(name::AbstractString) = PackageSpec(name)
 Package(name::AbstractString, uuid::UUID) = PackageSpec(name, uuid)
 Package(name::AbstractString, uuid::UUID, version::VersionTypes) = PackageSpec(name, uuid, version)
 
+
+"""
+    @__VERSION__ -> Union{VersionNumber, Nothing}
+
+Get the `VersionNumber` of the package which expands this macro. If executed outside of a
+package `nothing` will be returned.
+"""
+macro __VERSION__()
+    pkg_dir = Base.pkgdir(__module__)
+
+    if pkg_dir !== nothing
+        project_data = TOML.parsefile(joinpath(pkg_dir, "Project.toml"))
```

Comment:
`joinpath(pkg_dir, "Project.toml"))` should be `Types.projectfile_path(pkg_dir)` because `Project.toml` is not the only name for a project file